store/models/eth.go

```diff
@@ -95,16 +95,61 @@ func (f *FunctionSelector) UnmarshalJSON(input []byte) error {
 	return nil
 }
 
-// BlockHeader is the parameters passed in notifications for new blocks.
+// Represents a block header in the Ethereum blockchain.
+// Deliberately does not have required fields because some fields aren't
+// present depending on the Ethereum node.
+// i.e. Parity does not always send mixHash
 type BlockHeader struct {
-	Number hexutil.Big `json:"number" storm:"id,index,unique"`
+	ParentHash  common.Hash      `json:"parentHash"`
+	UncleHash   common.Hash      `json:"sha3Uncles"`
```

Comment:
My motivation behind including the additional BlockHeader fields is to convey its intent of storing header fields. When simply having `Number` it's easy to want to mold this into something specific for `HeadTracker`, `storm` or `boltdb`. Having `BlockHeader` with just `Number` and then adding `IndexableBlockNumber` seems redundant without this.